firebase-common/src/commonMain/kotlin/dev/gitlive/firebase/decoders.kt

```diff
@@ -122,18 +131,35 @@ open class FirebaseCompositeDecoder constructor(
         previousValue: T?
     ): T? {
         val isNullabilitySupported = deserializer.descriptor.isNullable
-        return if (isNullabilitySupported || decodeNotNullMark(get(descriptor, index))) decodeSerializableElement(descriptor, index, deserializer, previousValue) else decodeNull(get(descriptor, index))
+        return if (isNullabilitySupported || decodeElement(descriptor, index, ::decodeNotNullMark)) {
+            decodeSerializableElement(descriptor, index, deserializer, previousValue)
+        } else {
+            decodeElement(descriptor, index, ::decodeNull)
+        }
     }
 
-    override fun decodeShortElement(descriptor: SerialDescriptor, index: Int) = decodeShort(get(descriptor, index))
+    override fun decodeShortElement(descriptor: SerialDescriptor, index: Int) =
+        decodeElement(descriptor, index, ::decodeShort)
 
-    override fun decodeStringElement(descriptor: SerialDescriptor, index: Int) = decodeString(get(descriptor, index))
+    override fun decodeStringElement(descriptor: SerialDescriptor, index: Int) =
+        decodeElement(descriptor, index, ::decodeString)
 
     override fun endStructure(descriptor: SerialDescriptor) {}
 
     @ExperimentalSerializationApi
     override fun decodeInlineElement(descriptor: SerialDescriptor, index: Int): Decoder =
-        FirebaseDecoder(get(descriptor, index))
+        decodeElement(descriptor, index, ::FirebaseDecoder)
+
+    private fun <T> decodeElement(descriptor: SerialDescriptor, index: Int, decoder: (Any?) -> T): T {
+        return try {
+            decoder(get(descriptor, index))
+        } catch (e: Exception) {
```

Comment:
`} catch (exception: Exception) {` or `} catch (error: Exception) {` will be better

Reply:
+1 for `exception`